acceptance/launcher_test.go

```diff
@@ -282,6 +282,22 @@ func testLauncher(t *testing.T, when spec.G, it spec.S) {
 				)
 			})
 
+			it("merges env vars from system, user provided and CNB", func() {
+				if runtime.GOOS == "windows" {
+					cmd := exec.Command("docker", "run", "--rm",
+						`--env=PATH=c:\Windows\system32;c:\extra`,
+						launchImage, "--",
+						`cmd`, `/c`, `echo %PATH%`)
+					assertOutput(t, cmd, `c:\layers\some_buildpack\some_layer\bin;C:\Windows\system32;C:\Windows;c:\extra`)
+				} else {
+					cmd := exec.Command("docker", "run", "--rm",
+						"--env=PATH=/bin:/extra",
+						launchImage, "--",
+						"sh", "-c", "echo $PATH")
+					assertOutput(t, cmd, "/layers/some_buildpack/some_layer/bin:/bin:/extra")
+				}
+			})
+
```

Comment:
I tried to include some acceptance-level coverage. Not 100% it conveys the purpose but I feel like it would catch any regressions.